pallets/accounts/src/lib.rs

```diff
@@ -1090,12 +1122,147 @@ pub mod pallet {
 			Ok(())
 		}
 
-		fn apply_20_percent(amount: u128) -> u128 {
+		/// Withdraw vested tickets and NOVA
+		fn withdraw_tickets_and_nova(account: T::AccountId) -> DispatchResult {
+			const SECONDS_IN_WEEK: u64 = 3600 * 24 * 7;
+			const SECONDS_IN_BLOCK: u64 = 6;
+			let current_block_number =
+				<frame_system::Pallet<T>>::block_number().saturated_into::<u64>();
+
+			let eth_account = <EVMLinks<T>>::get(&account);
+			if let Some(eth_account) = eth_account {
+				let eth_locked_frag = <EthLockedFrag<T>>::get(&eth_account);
+				if let Some(eth_locked_frag) = eth_locked_frag {
+					// Retrieve the info related to the FRAG locked by this account
+					let frag_lock_block_number =
+						eth_locked_frag.first_lock_block_number.saturated_into::<u64>();
+					let total_frag_locked_on_eth = eth_locked_frag.amount.saturated_into::<u64>();
+					let frag_lock_period = eth_locked_frag.lock_period.saturated_into::<u64>();
+
+					// convert for convenience. The calculations below are made by num of weeks
+					// calculated from the num of blocks from the block num of the FRAG lock to the current block.
+					let lock_period_in_weeks =
+						Self::eth_lock_period_to_weeks(frag_lock_period).unwrap(); // unwrap here is safe
+
+					// Example: 100 FRAG locked => <80 FRAG convertible to Tickets immediately, 20 vested>
+					// If vesting period = 4 weeks => 20 / 4 = 5 FRAG convertible each week.
+					let tickets_convertible_per_week = Self::apply_percent(
+						total_frag_locked_on_eth,
+						100 - T::InitialPercentageTickets::get(),
+					) / lock_period_in_weeks;
+
+					let nova_convertible_per_week = Self::apply_percent(
+						total_frag_locked_on_eth,
+						100 - T::InitialPercentageNova::get(),
+					) / lock_period_in_weeks;
+
+					let current_frag_price = Self::get_oracle_price();
+					// The amount of Tickets and NOVA will depend on the price of 1 FRAG at the time of withdraw
+					let tickets_amount_per_week_at_current_price = current_frag_price *
+						T::USDEquivalentAmount::get() *
+						tickets_convertible_per_week;
+					let nova_amount_at_current_price = current_frag_price *
+						T::USDEquivalentAmount::get() *
+						nova_convertible_per_week;
+
+					let num_weeks_since_lock_frag = (current_block_number - frag_lock_block_number) // The blocks passed from FRAG lock
+							* SECONDS_IN_BLOCK / SECONDS_IN_WEEK +
+						1; // converted to weeks
+					let tickets_amount_to_withdraw =
+						tickets_amount_per_week_at_current_price * num_weeks_since_lock_frag;
+					let nova_amount_to_withdraw =
+						nova_amount_at_current_price * num_weeks_since_lock_frag;
+
+					sp_std::if_std! {
+						// This code is only being compiled and executed when the `std` feature is enabled.
+						println!("frag_lock_block_num: {}", frag_lock_block_number);
```

Comment:
uh? why not use `log::trace!` works in whatever mode

Reply:
I wanted to see logs in the console while testing, i will remove all this once i reopen this PR, i never intended to leave this in the final code.